internal/auth0/client/expand.go

```diff
@@ -236,6 +236,44 @@ func expandClientMetadata(d *schema.ResourceData) *map[string]interface{} {
 	return &newMetadataMap
 }
 
+func expandClientAddons(d *schema.ResourceData) *management.ClientAddons {
+	if !d.HasChange("addons") {
+		return nil
+	}
+
+	var addons management.ClientAddons
+
+	d.GetRawConfig().GetAttr("addons").ForEachElement(func(_ cty.Value, addonsCfg cty.Value) (stop bool) {
+		addons.AWS = expandClientAddonAWS(addonsCfg.GetAttr("aws"))
+
+		return stop
+	})
+
+	if addons == (management.ClientAddons{}) {
+		return nil
+	}
+
+	return &addons
+}
+
+func expandClientAddonAWS(awsCfg cty.Value) *management.AWSClientAddon {
+	var awsAddon management.AWSClientAddon
+
+	awsCfg.ForEachElement(func(_ cty.Value, awsCfg cty.Value) (stop bool) {
+		awsAddon.Principal = value.String(awsCfg.GetAttr("principal"))
+		awsAddon.Role = value.String(awsCfg.GetAttr("role"))
+		awsAddon.LifetimeInSeconds = value.Int(awsCfg.GetAttr("lifetime_in_seconds"))
```

Comment:
656I see this was updated in #656 https://github.com/auth0/terraform-provider-auth0/blob/52cdcf0a944dbfd714af2179573305ece61b8961/internal/auth0/client/expand.go#L263

Reply:
Yeah, I was playing around with a new style in the second PR to try to circumvent the API behavior when you switch from one addon to another one. Both are equivalent but I think we'll go forward with the ones from the second PR.